public/app/features/dashboard/components/AddPanelButton/AddPanelButton.tsx

```diff
@@ -11,12 +11,19 @@ import AddPanelMenu from './AddPanelMenu';
 
 export interface Props {
   dashboard: DashboardModel;
+  onToolbarAddClick?: () => void;
 }
 
-const AddPanelButton = ({ dashboard }: Props) => {
+const AddPanelButton = ({ dashboard, onToolbarAddClick }: Props) => {
   const styles = getStyles(useTheme2());
   const [isMenuOpen, setIsMenuOpen] = useState(false);
 
+  useEffect(() => {
+    if (isMenuOpen && onToolbarAddClick) {
+      onToolbarAddClick();
```

Comment:
shouldn't it be something like `onToolbarAddOpen` or `onToolbarAddMenuOpen`  ?
I mean, it sounds like this function will be executed when add button is clicked, instead of when the menu is open